configure.ac

```diff
@@ -1385,14 +1385,13 @@ fi
 
 if test "$use_usdt" != "no"; then
   AC_MSG_CHECKING([whether Userspace, Statically Defined Tracing tracepoints are supported])
-  AC_COMPILE_IFELSE([
-    AC_LANG_PROGRAM(
-      [#include <sys/sdt.h>],
-      [DTRACE_PROBE("context", "event");]
-    )],
-    [AC_MSG_RESULT([yes]); AC_DEFINE([ENABLE_TRACING], [1], [Define to 1 to enable tracepoints for Userspace, Statically Defined Tracing])],
-    [AC_MSG_RESULT([no]); use_usdt=no;]
-  )
+  AC_CHECK_HEADER([sys/sdt.h], [use_usdt=yes], [use_usdt=no])
+else
+  use_usdt=no
+fi
+
+if test x$use_usdt = xyes; then
+  AC_DEFINE([ENABLE_TRACING], [1], [Define to 1 to enable tracepoints for Userspace, Statically Defined Tracing])
```

Comment:
If this PR is macOS-specific, why remove Linux-specific check for `DTRACE_PROBE` macro?

Reply:
Before https://github.com/bitcoin/bitcoin/pull/22238 `AC_CHECK_HEADER([sys/sdt.h], [have_sdt=yes], [have_sdt=no])`was the check to detect whether USDT tracepoints are supported, but per the reasoning of the aformentioned PR, the indent of that change was to "fail" on macOS and not on Linux as macOS has the `sys/sdt.h` header but `DTRACE_PROBE` is not defined in it.

So even after this macOS-specific PR, if I did not change that, the binary would compiled with USDT tracing  disabled as the `DTRACE_PROBE` is still not defined. That's why I revert that change, as now there is a macOS specific clause in `util/trace.h`

Reply:
For reference, my understanding of the effects of that PR was wrong, as it seems that having `sys/sdt.h` header but not defined `DTRACE_PROBE` is not a macOS-specific pattern. I have since reverted this change.